Update abapEnvironmentAddons.md: Refine branch change description (#3274)
SAP/jenkins-library: documentation/docs/scenarios/abapEnvironmentAddons.md

- `commitID`: this is the commitID from the git repository
- `languages`: specify the languages to be delivered according to ISO-639. For all deliveries of an Add-on Product Version, the languages should not change. If languages should be added, a new Add-on Product Version must be created.

`addonVersion` influences solely the creation of the target vector. Without target vector nothing can be deployed. But it is possible to deploy combinations which have been build in the past (especially if the same software component version is part of multiple addon products).

> The `version`of software components influcences two aspects:

Reviewer: ```suggestion
The `version` of a software component influcences two aspects:
```

- The given version will be used as part of the target vector
- If there exists NO delivery package with the given version in AAKaaS the build of this package is performed

> As a result, if `addonVersion` is increased but references only to software component (repository) `version` for which a delivery package has already been build in the past no new package is build, only a new target vector is created.

Reviewer: ```suggestion
As a result, if the `addonVersion` is increased but references a software component (repository) `version` for which a delivery package has already been built no new package is build. Only a new target vector is created.
```

Changing the `addonVersion` string does not necessarily imply that new delivery packages are being created. In case software component versions are used that were already part of a previous add-on `addonVersion`, the existing delivery packages are reused for the new add-on product version.
Only by changing the `version` of a software component, the build of a new delivery package with the latest changes is triggered.

> The `addonVersion` should be determined synchronously to how the software components bundle is changed: In case the release version of a software component is changed, the release of the `addonVersion` should be changed. If the support package version of a software component is changed, support package version of the add-on should be changed. And if patch version of a software component is changed, the patch version of the add-on should be adjusted.

Reviewer: ```suggestion
The `addonVersion` should be determined similarly to how the software components bundle is changed: In case the release version of a software component is changed, the release of the `addonVersion` should be changed. If the support package version of a software component is changed, support package version of the add-on should be changed. And if patch version of a software component is changed, the patch version of the add-on should be adjusted.
```
